drivers/input/input_adc_keys.c

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright (c) 2024 Chen Xingyu <hi@xingrz.me>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT adc_keys
+
+#include <stdlib.h>
+#include <stdbool.h>
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/adc.h>
+#include <zephyr/input/input.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/util.h>
+
+LOG_MODULE_REGISTER(adc_keys, CONFIG_INPUT_LOG_LEVEL);
+
+struct adc_keys_code_config {
+	uint32_t key_index;
+	int32_t press_mv;
+};
+
+struct adc_keys_key_state {
+	bool last_state;
+	bool curr_state;
+};
+
+struct adc_keys_config {
+	struct adc_dt_spec channel;
+	uint32_t sample_period_ms;
+	int32_t keyup_mv;
+	const struct adc_keys_code_config *code_cfg;
+	uint32_t code_cnt;
```

Comment:
this and `key_count` maybe could be 16 or 8 bits?

Reply:
Updated. Using `uint8_t` supports up to 8 keys (2^8-1=255 combinations) with a minimum 13mV difference between combinations. This should be sufficient.